xla/mlir_hlo/mhlo/IR/hlo_ops.td

```diff
@@ -1451,12 +1451,16 @@ def MHLO_WhileOp: MHLO_Op<"while", [
   let hasVerifier = 1;
 }
 
-def MHLO_AllGatherOp : MHLO_Op<"all_gather", [SameOperandsAndResultElementType]> {
+def MHLO_AllGatherOp : MHLO_Op<"all_gather", [
+    SameOperandsAndResultElementType,
+    SingleBlockImplicitTerminator<"ReturnOp">
```

Comment:
The SingleBlockImplicitTerminator trait is not needed here. Unlike all-reduce op which has an attached region for the reduction computation, AllGather does not, so this trait is not relevant.

Reply:
fixed